### PR TITLE
Update Angular Material and CLI

### PR DESCRIPTION
**Pull request by prydonius**
Updating to the latest version of CLI which fixes AOT build failures
Updating Material to the latest version which removes MaterialModule (requiring individual modules to be manually imported) and also switches to the "Mat" prefix since "Md" is now deprecated.

See https://github.com/angular/material2/releases for more info.

### CODE REVIEW

**Comment by prydonius** [comment]
@kemcake this change is causing some styling issues, would you be able to take this over and fix these?

screen shot 2017-09-26 at 11 27 30
screen shot 2017-09-26 at 11 27 22

File: src/ui/src/app/chart-details/chart-details-usage/chart-details-usage.component.html [hunk 1]
```diff
@@ -1,37 +1,41 @@
...
-      <mat-tab label="Helm CLI">
-        <div *ngIf=showRepoInstructions class="chart-details-usage__repository">
-          <h2 class="chart-details-usage__label">Add {{ chart.attributes.repo.name }} repository</h2>
+    <mat-tab *ngIf="config.releasesEnabled" label="One Click installation">
+      <button color="primary" mat-raised-button
+       (click)="installDeployment(chart.id, currentVersion)"
+       *ngIf="!installing">
+        Install {{ chart.attributes.name }} v{{ currentVersion }}
+      </button>
+      <p *ngIf="installing"><mat-progress-bar mode="indeterminate"></mat-progress-bar></p>
```
Comment (prydonius) [inline src/ui/src/app/chart-details/chart-details-usage/chart-details-usage.component.html]: Is progress bar another module we need to import?

File: src/ui/src/app/main-header/main-header.component.html [hunk 2]
```diff
@@ -8,7 +8,7 @@
     </div>
     <div class="main-header-search">
       <form (ngSubmit)="searchCharts(searchBox)">
-        <mat-form-field class="md-input--reverse md-input--full md-input--big">
+        <mat-form-field class="mat-input--reverse mat-input--full mat-input--big">
```
Comment (prydonius) [inline src/ui/src/app/main-header/main-header.component.html]: ah I see, I guess the https://www.npmjs.com/package/angular-material-prefix-updater tool I used didn't account for class modifiers!

Comment (kemcake) [reply src/ui/src/app/main-header/main-header.component.html]: This is custom css classes we used so yes I guess the updater is not so magic haha

**Comment by prydonius** [comment]
thanks @kemcake!

**Pull request merged**